packages/ledgerstate/color.go

```diff
@@ -0,0 +1,228 @@
+	if color, err = ColorFromMarshalUtil(marshalUtil); err != nil {
+		err = xerrors.Errorf("failed to parse Color from MarshalUtil: %w", err)
+		return
+	}
+	consumedBytes = marshalUtil.ReadOffset()
+
+	return
+}
+
+// ColorFromBase58EncodedString creates a Color from a base58 encoded string.
+func ColorFromBase58EncodedString(base58String string) (color Color, err error) {
+	bytes, err := base58.Decode(base58String)
+	if err != nil {
+		err = xerrors.Errorf("error while decoding base58 encoded Color (%v): %w", err, ErrBase58DecodeFailed)
+		return
+	}
+
+	if color, _, err = ColorFromBytes(bytes); err != nil {
+		err = xerrors.Errorf("failed to parse Color from bytes: %w", err)
+		return
+	}
+
+	return
+}
+
+// ColorFromMarshalUtil unmarshals a Color using a MarshalUtil (for easier unmarshaling).
+func ColorFromMarshalUtil(marshalUtil *marshalutil.MarshalUtil) (color Color, err error) {
+	colorBytes, err := marshalUtil.ReadBytes(ColorLength)
+	if err != nil {
+		err = xerrors.Errorf("failed to parse Color (%v): %w", err, ErrParseBytesFailed)
+		return
+	}
+	copy(color[:], colorBytes)
+
+	return
+}
+
+// Bytes marshals the Color into a sequence of bytes.
+func (c Color) Bytes() []byte {
+	return c[:]
+}
+
+// Base58 returns a base58 encoded version of the Color.
+func (c Color) Base58() string {
+	return base58.Encode(c.Bytes())
+}
+
+// String creates a human readable string of the Color.
+func (c Color) String() string {
+	switch c {
+	case ColorIOTA:
+		return "IOTA"
+	case ColorMint:
+		return "MINT"
+	default:
+		return c.Base58()
+	}
+}
+
+// ColorIOTA is the zero value of the Color and represents uncolored tokens.
+var ColorIOTA = Color{}
+
+// ColorMint represents a placeholder Color that indicates that tokens should be "colored" in their Output.
+var ColorMint = Color{255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255}
+
+// endregion ///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// region ColoredBalances //////////////////////////////////////////////////////////////////////////////////////////////
+
+// ColoredBalances represents a collection of balances associated to their respective Color that maintains a
+// deterministic order of the present Colors.
+type ColoredBalances struct {
+	balances *orderedmap.OrderedMap
+}
+
+// NewColoredBalances returns a new deterministically ordered collection of ColoredBalances.
+func NewColoredBalances(balances map[Color]uint64) (coloredBalances *ColoredBalances) {
+	coloredBalances = &ColoredBalances{balances: orderedmap.New()}
+
+	// deterministically sort colors
+	sortedColors := make([]Color, 0, len(balances))
+	for color := range balances {
+		sortedColors = append(sortedColors, color)
+	}
+	sort.Slice(sortedColors, func(i, j int) bool { return bytes.Compare(sortedColors[i][:], sortedColors[j][:]) < 0 })
```

Comment:
Maybe it'd be worth to fulfil the `sort.Interface` for `Color` so this could just be a `sort.Sort`.

Reply:
This would require us to introduce sth like `type Colors []Color` so I think it doesn't really make sense